### PR TITLE
Feature/build containerized app

### PR DESCRIPTION
**Pull request by Nick0603**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Create Dockerfile for development and production.
Create Deploy pipeline in stage and main branch.
Update README.md

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Create Development
- [x] Run the `docker-compose up -d` And all the service will be created.

Deploy
- [x] Push to stage branch and the stage site will be updated. (https://stage.nick-chen-sudo.com/)
- [x] Push to main branch and the stage site will be updated. (https://nick-chen-sudo.com/)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push Docker image
+        id: build_image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          tags: nickchensudo/itseed_backend:main-${{ github.sha }}
```
Comment (AlanSyue) [inline .github/workflows/main.yaml:28]: Does it need to change `nickchensudo` to `itseed` ?

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ap-northeast-1
+
+      - name: Fill in the new image ID in the Amazon ECS task definition
+        id: task-def
+        uses: aws-actions/amazon-ecs-render-task-definition@v1
+        with:
+          task-definition: task-definition--production.json
+          container-name: itseed-backend
+          image: nickchensudo/itseed_backend:main-${{ github.sha }}
```
Comment (AlanSyue) [inline .github/workflows/main.yaml:53]: Same as:
https://github.com/ITSeed-Official/itseed_backend/pull/5/commits/b403829980ca8f99d8d41e92344413e13c5080f1#r712826592

File: .github/workflows/stage.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push Docker image
+        id: build_image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          tags: nickchensudo/itseed_backend:stage
```
Comment (AlanSyue) [inline .github/workflows/stage.yaml:28]: Same as
https://github.com/ITSeed-Official/itseed_backend/pull/5/commits/b403829980ca8f99d8d41e92344413e13c5080f1#r712826592

File: .github/workflows/main.yaml (hunk 1 above)
Comment (Nick0603) [reply .github/workflows/main.yaml:28]: That's a docker account name. We need to create a new docker account for the project.

Comment (Nick0603) [reply .github/workflows/main.yaml]: I change the docker id to itseedsystem

**Review by AlanSyue — APPROVED** [review]

**Pull request merged**